### PR TITLE
Add Versioneer

### PR DESCRIPTION
**Pull request by mikemhenry**
Fixes issue #23 

We need to decide how we want to tag releases 
https://github.com/python-versioneer/python-versioneer/blob/master/INSTALL.md
so we can set `tag_prefix` correctly.

```
tag_prefix:

a string, like 'PROJECTNAME-', which appears at the start of all VCS tags. If your tags look like 'myproject-1.2.0', then you should use tag_prefix='myproject-'. If you use unprefixed tags like '1.2.0', this should be an empty string, using either tag_prefix= or tag_prefix=''.
```

### CODE REVIEW

**Comment by mikemhenry** [comment]
Do we want to have pep8speaks ignore `openfe/_version.py` and `versioneer.py` or should I fix them (easy with black or something)

File: setup.cfg [hunk 1]
```diff
@@ -35,3 +35,10 @@ test =
 
 [bdist_wheel]
 universal = 1
+
+[versioneer]
+VCS = git
+style = pep440-post
+versionfile_source = openfe/_version.py
+versionfile_build = openfe/_version.py
+tag_prefix = "" # see https://github.com/python-versioneer/python-versioneer/blob/master/INSTALL.md
```
Comment (IAlibay) [inline setup.cfg:44]: This is causing issues
```suggestion
tag_prefix = '' # see https://github.com/python-versioneer/python-versioneer/blob/master/INSTALL.md
```

**Review by IAlibay — COMMENTED** [review]
Couple comments

**Comment by IAlibay** [comment]
> Do we want to have pep8speaks ignore `openfe/_version.py` and `versioneer.py` or should I fix them (easy with black or something)

So I believe for the most part pep8speaks shouldn't complain beyond this one PR (unless we change something). But if you feel like autoformatting it shouldn't have any impact (unless we need to patch the file for some reason in the future)?

File: setup.cfg (hunk 1 above)
Comment (IAlibay) [reply setup.cfg:44]: Actually I wonder if it's more the comment afterwards that is causing issues (given it's attempted to add quotes around it)

**Comment by richardjgowers** [comment]
not too fussed on version string style, "v1.2.3" is ok, I'm not offended if we include the package name though "openfe-v1.2.3"

File: setup.cfg (hunk 1 above)
Comment (IAlibay) [reply setup.cfg:44]: Sorry I 100% missed that this was probably intentional 🤦🏽‍♂️

File: setup.py [hunk 2]
```diff
@@ -1,2 +1,8 @@
 from setuptools import setup
-setup()
+
+import versioneer
+
+setup(
+    version=versioneer.get_version(),
```
[the remark this answers is not in the document]
Comment (mikemhenry) [reply setup.py:6]: I think it is needed per the installation instructions: https://github.com/python-versioneer/python-versioneer/blob/master/INSTALL.md

```
add a import versioneer to your setup.py, and add the following arguments to the setup() call:

  version=versioneer.get_version(),
  cmdclass=versioneer.get_cmdclass(),
```

**Comment by dwhswenson** [comment]
Re: pep8speaks. I'd vote for adding the ignore. It's a vendored file; I usually try to leave those as-is if possible. Keeps sane diffs for updates, etc. Also, any future updates shouldn't trigger pep8speaks.

File: setup.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mikemhenry) [reply setup.py:6]: oops, you said `setup.cfg` yes that needs to go!

File: setup.cfg [hunk 3]
```diff
@@ -35,3 +35,10 @@ test =
 
 [bdist_wheel]
 universal = 1
+
```
[the remark this answers is not in the document]
Comment (mikemhenry) [reply setup.cfg:37]: Good catch!

File: setup.cfg (hunk 1 above)
Comment (mikemhenry) [reply setup.cfg]: Haha, I left that comment in there to remind myself when I made the PR to ask about it. I will fix it.

**Comment by mikemhenry** [comment]
If pep8speaks doesn't complain about the files again, then I won't bother adding them to an ignore file. RE tagging, I'm fine with just tagging the version number, I'm not sure what value adding the `openfe` prefix brings to the tag.

**Comment by IAlibay** [comment]
> If pep8speaks doesn't complain about the files again, then I won't bother adding them to an ignore file. RE tagging, I'm fine with just tagging the version number, I'm not sure what value adding the `openfe` prefix brings to the tag.

Yeah pretty sure that by default it only acts on diffs, see: https://github.com/OrkoHunter/pep8speaks/blob/master/data/default_pep8speaks.yml

**Comment by richardjgowers** [comment]
thanks @mikemhenry !

**Pull request merged**